src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[4];
+
+        for (int i = 0; i < figures.length / 2; i++) {
```

Comment:
The loop correctly uses `figures.length / 2` to fill half of the array with random figures.